### DISCUSSION TITLE
Re-think compiler options for c++ standard selection.

### DISCUSSION THREAD

**Discussion by bmanga** [open-watcom/open-watcom-v2#330] [Ideas]
After a discussion with @pchapin, we came to the conclusion that the *-za0x* is not the best option to enable c++11, as it is both confusing (does it also enable the *-za* flag?) and not future proof (what about c++14?).

We considered either enabling *-za0x* it by default and removing it (thus supporting only an "open-watcom standard"), or moving to something like gcc/clang 's *-std=c++xy* .

The first approach may cause frictions due to the list of breaking changes that c++11 introduced, and enabling it by default will break some code.

The second is widely used, but there is little benefit for ow at the moment, as it doesn't support any standard yet.

I would prefer the second option, but I would like to know other people's views on the matter.

**Comment by pchapin** [comment]
As I mentioned in a different issue, my personal preference would be to just support a single "standard," even if only incompletely. The breaking changes in C++ 2011 strike me as unlike to cause real problems for Open Watcom users (they are either minor or easily fixed), and I feel like the cost of maintaining multiple standards would be too high.

That said, I appreciate @bmanga's points and I could get behind a decision to support multiple standards. I understand right now this means C++ 1998, which we mostly support already, and C++ 2011, which we barely support at all (at the moment). Setting the stage for other standards in the future makes sense in this context.

So, officially I vote for the first option, but I am fine with the second option also and would support it if that's what others think is best.

**Comment by jmalak** [comment]
I think we should have two mode one legacy (existing) and new C++ 2011 standard.
If we do development of C++ compiler parallel to existing version then it is possible to do it for only one standard. To be C++ 2011 compliant will require massive changes in existing compiler that do it on existing version could be harder than in parallel way and we can deside when old C++ compiler will be replaced by new one.

**Comment by pchapin** [comment]
@jmalak, are you suggesting that we should work on C++11 support in a branch?

**Comment by jmalak** [comment]
It is question for discussion.
